### PR TITLE
Verified user badge

### LINKED ISSUE #3684: [Task] Show in user's profile when they have been verified
Please see and discuss the details in the meta issue.

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Show a "Verified" badge in room member detail page when the user is verified. If the user is not verified, show a disabled action to inform the user that they must use another client to verify a user.

First commits are extraction of UI component `MatrixBadge` and creation of `MatrixBadgeRowMolecule`.

## Motivation and context

Closes #3684 

## Screenshots / GIFs

See recorded one

## Tests

- Open room member page for verified user and not verified user and observe the content.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/features.userprofile.shared_UserProfileView_Day_0_en.png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/features.userprofile.shared_UserProfileView_Day_0_en.png:1]: Oups, let me check what happened here (I am rebasing too many times this branch...)

File: features/userprofile/shared/src/main/kotlin/io/element/android/features/userprofile/shared/UserProfileStateProvider.kt [hunk 1]
```diff
@@ -20,13 +20,12 @@ open class UserProfileStateProvider : PreviewParameterProvider<UserProfileState>
...
             aUserProfileState(userName = null),
-            aUserProfileState(isBlocked = AsyncData.Success(true)),
+            aUserProfileState(isBlocked = AsyncData.Success(true), isVerified = AsyncData.Success(true)),
             aUserProfileState(displayConfirmationDialog = UserProfileState.ConfirmationDialog.Block),
             aUserProfileState(displayConfirmationDialog = UserProfileState.ConfirmationDialog.Unblock),
-            aUserProfileState(isBlocked = AsyncData.Loading(true)),
+            aUserProfileState(isBlocked = AsyncData.Loading(true), isVerified = AsyncData.Loading()),
             aUserProfileState(startDmActionState = AsyncAction.Loading),
             aUserProfileState(canCall = true),
-            aUserProfileState(dmRoomId = null),
```
Comment (bmarty) [inline features/userprofile/shared/src/main/kotlin/io/element/android/features/userprofile/shared/UserProfileStateProvider.kt:29]: This state was the same than the first one, so I removed it.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/encryption/RustEncryptionService.kt [hunk 2]
```diff
@@ -204,8 +205,18 @@ internal class RustEncryptionService(
...
     override suspend fun pinUserIdentity(userId: UserId): Result<Unit> = runCatching {
-        val userIdentity = service.userIdentity(userId.value) ?: error("User identity not found")
-        userIdentity.pin()
+        getUserIdentity(userId).pin()
+    }
+
+    private suspend fun getUserIdentity(userId: UserId): UserIdentity {
+        return service.userIdentity(
+            userId = userId.value,
+            // requestFromHomeserverIfNeeded = true,
```
Comment (bmarty) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/encryption/RustEncryptionService.kt:219]: This param will come later.

Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/encryption/RustEncryptionService.kt:219]: Actually no, I'll cleanup later.

**Review by ganfra — APPROVED** [review]
Some small remarks otherwise LGTM!

File: features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsPresenter.kt [hunk 3]
```diff
@@ -112,6 +113,21 @@ class RoomDetailsPresenter @Inject constructor(
 
         val roomNotificationSettingsState by room.roomNotificationSettingsStateFlow.collectAsState()
 
+        val roomBadges by produceState(persistentListOf(), isPublic) {
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsPresenter.kt:116]: > If the room becomes encrypted this won't be refreshed.

I have seen that. We need `isEncrypted` property to be added in RoomInfo for fix that properly across the application.

[the remark this answers is not in the document]
Comment (bmarty) [reply features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsPresenter.kt:116]: Done in 35867f9837e993fab4e0da4c29abeea5747729b2

File: features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsView.kt [hunk 4]
```diff
@@ -401,36 +402,41 @@ private fun ColumnScope.TitleAndSubtitle(
...
-                )
-            }
-            if (isPublic) {
-                RoomBadge.View(
-                    text = stringResource(R.string.screen_room_details_badge_public),
-                    icon = CompoundIcons.Public(),
-                    type = RoomBadge.Type.Neutral,
-                )
-            }
+    if (roomBadge.isEmpty()) return
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsView.kt:408]: 2eec17fec7e630239abddadcdeec6fde22c47127

**Pull request merged**